tests/php/sync/test_class.jetpack-sync-posts.php

```diff
@@ -630,6 +630,9 @@ function test_filters_out_blacklisted_post_types() {
 		$this->assertFalse( $this->server_replica_storage->get_post( $post_id ) );
 		$sync_event = $this->server_event_storage->get_most_recent_event( 'jetpack_sync_save_post' );
 		$this->assertFalse( $sync_event );
+
+		// Clean up.
+		unregister_post_type( 'snitch' );
```

Comment:
we actually need to move the unregister_post_type before the assertions. Otherwise if this test fails it will still be registered and disrupt other tests.

Reply:
I'm late for this patch, but another pattern for that sort of thing is to do your code like
```
register_post_type( 'snitch', $args );
try {
	// All your code in here
} finally {
	// Clean up.
	unregister_post_type( 'snitch' );
}
```
That'll also do the right thing if some other code you call throws unexpectedly.

Reply:
Since then I also found [`reset_post_types()`](https://github.com/WordPress/wordpress-develop/blob/d6fa3df42031c493ce6c4b4ff5f32be1745067d1/tests/phpunit/includes/abstract-testcase.php#L263), which Core runs for its tests on `setUp`.